e2etest/pkg/config/validate.go

```diff
@@ -14,6 +14,13 @@ import (
 	"k8s.io/kubernetes/test/e2e/framework"
 )
 
+func ValidateAnnotateIPAllocatedFromPool(addressPool *metallbv1beta1.IPAddressPool, annotations map[string]string) error {
+	if annotations["metallb.universe.tf/ip-allocated-from-pool"] == addressPool.Name {
```

Comment:
Given how much we simplified this, I think we can use the if directly and get rid of the function

Reply:
Thanks!